salt/modules/disk.py

```diff
@@ -481,12 +481,13 @@ def fstype(device):
                 fs_type = df_out[2]
                 if fs_type:
                     return fs_type
-        else:
-            df_out = __salt__['cmd.run']('df -T {0}'.format(device)).splitlines()
-            if len(df_out) > 1:
-                fs_type = df_out[1]
-                if fs_type:
-                    return fs_type
```

Comment:
What is the point in removing this, if it won't find the filetype, and just continue on?

Wouldn't it be better to only use the `file -s` if the disk is unmounted?